Standards/scs-xxxx-v1-ipv4-networking-standard.md

```diff
@@ -0,0 +1,79 @@
+---
+title: IPv4 Networking Standard
+type: Standard
+status: Proposal
+track: IaaS
+---
+
+## Introduction
+
+This document outlines the standardized approach for the management and allocation of
+public IPv4 addresses within Sovereign Cloud Stack (SCS) environments. Its aim is to
+ensure a consistent, secure, and efficient methodology for IP address provisioning
+across all SCS cloud services.
+
+## Motivation
+
+The motivation behind establishing this standard is to enhance interoperability, improve
+security measures, and streamline the operational processes across different SCS clouds.
+It addresses the need for a unified procedure in handling IPv4 networking to facilitate
+ease of use for IaaS users and DevOps teams.
+
+## Design Considerations
+
+### Options considered
+
+#### _Neutron Routers_
+
+Usage of Neutron Routers: To manage traffic between internal and external networks Neutron Routers **MUST** be used as the default gateway for VMs requiring access to external networks and the internet, thereby facilitating the routing of traffic and enhancing network security.
+
+CSPs **SHOULD** use OVN or L3agent as High Availability (HA) service deployments.
+Standard external networks **MUST NOT** be made accessible as _shared networks_. It is advised that external networks are only reachable by the usage of routing and floating IPs.
+However, for special use cases like certain storage or VPN solutions it could be useful to allow _direct access networks_.
+
+External networks and subnets **SHOULD** (very strong should) be configured with _--no-dhcp_ (DHCP - Dynamic Host Configuration Protocol). It is more secure to configure it like this, since it gives less space for reflection attacks, e.g. _Denial of Service_ (DOS) attacks. If _dhcp_ is configured, certain firewall configurations **MUST** be made to catch IPs from the _Neutron dhcp agent_ in the public network.
+
+#### _Neutron Plugins_
+
+Neutron Plugins: A SCS conform CSP **MAY** use RBAC and VPNaaS plugins.
```

Comment:
Does a current list of plugins exist? When I look into the code [^1] I only see the ML2 Plugin. Also the documentation does not mention a plugin list (anymore). It seems to me, that most of it was removed or moved to `extensions`. When looking into the openstack repository on github, there are some deprecated plugins, a lot of charms-repos, some repos that could be plugins or agents and some other stuff [^2].
From working in secustack I know it is possible to add custom made Plugins.

I looked into my devstack config and it does not specify any plugins, except the ML2:
```
[DEFAULT]
service_plugins = ovn-router
rpc_state_report_workers = 0
api_workers = 2
notify_nova_on_port_data_changes = True
notify_nova_on_port_status_changes = True
auth_strategy = keystone
debug = True
core_plugin = ml2
dhcp_agent_notification = False
transport_url = rabbit://stackrabbit:MuchSecretSuchW0W@192.168.23.238:5672/
logging_exception_prefix = ERROR %(name)s ^[[01;35m%(instance)s^[[00m
logging_default_format_string = %(color)s%(levelname)s %(name)s [^[[00;36m-%(color)s] ^[[01;35m%(instance)s%(color)s%(message)s^[[00m
logging_context_format_string = %(color)s%(levelname)s %(name)s [^[[01;36m%(global_request_id)s %(request_id)s ^[[00;36m%(project_name)s %(user_name)s%(color)s] ^[[01;35m%(instance)s%(color>
logging_debug_format_suffix = ^[[00;33m{{(pid=%(process)d) %(funcName)s %(pathname)s:%(lineno)d}}^[[00m
bind_host = 0.0.0.0
use_syslog = False
state_path = /opt/stack/data/neutron
```

So all extensions seem to be usable all the time - because I was able to test the `network rbac`.

My conclusion: We should not state anything about plugins here, as they are poorly documented, not well maintained or even completely customized. We could discuss about letting CSPs add customized plugins. But all of these plugins while touching networking issues should not interfere with the scope of this standard. So exclude this, add a new issue and maybe we will find out enough for a new standard or maybe we just have a short note about Plugins overall on the docs-page.


[^1]: https://github.com/openstack/neutron/tree/master/neutron/plugins
[^2]: https://github.com/orgs/openstack/repositories?language=&q=neutron&sort=&type=all

Reply:
Some background for this as well: Neutron used to include a number of monolithic core plugins, but they have all been converted to drivers for the the ML2 core plugin, as well as service plugins that build on top of that.

The available service plugins are defined in neutrons [setup.cfg](https://github.com/openstack/neutron/blob/master/setup.cfg#L71), and CSPs can also configure additional external plugins there. Each plugin implementation has to declare the API extensions that it provides, most of which are defined in [neutron-lib](https://github.com/openstack/neutron-lib/tree/master/neutron_lib/api/definitions). The most prominent ones are `router` and `router-ovn` which implement virtual routers and floating IPs for l3agent and ovn, respectively.

A number of API extensions are also implemented by the ML2 core plugin itself, such as subnetpools, security groups, and the different rbac extensions. As such there is no separate plugin for RBAC.
There used to be a VPNaaS plugin in neutron but it has been removed at some point, though the definition for the API extension still exists in neutron-lib.